addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/handler/UniFiControllerThingHandler.java

```diff
@@ -0,0 +1,359 @@
+        Map<String, UniFiSite> siteMap = new HashMap<>();
+        UniFiSite[] sites = controller.getSites();
+        logger.debug("Found {} UniFi Site(s):", sites.length);
+        for (UniFiSite site : sites) {
+            logger.debug("  {}", site);
+            siteMap.put(site.getId(), site);
+        }
+        return siteMap;
+    }
+
+    private Map<String, UniFiDevice> getDevices() throws UniFiException {
+        Map<String, UniFiDevice> deviceMap = new HashMap<>();
+        Collection<UniFiSite> sites = sitesCache.values();
+        for (UniFiSite site : sites) {
+            Map<String, UniFiDevice> devices = getDevices(site);
+            deviceMap.putAll(devices);
+        }
+        return deviceMap;
+    }
+
+    private Map<String, UniFiDevice> getDevices(UniFiSite site) throws UniFiException {
+        Map<String, UniFiDevice> deviceMap = new HashMap<>();
+        UniFiDevice[] devices = controller.getDevices(site);
+        logger.debug("Found {} UniFi Device(s):", devices.length);
+        for (UniFiDevice device : devices) {
+            device.setSite(sitesCache.get(device.getSiteId()));
+            logger.debug("  {}", device);
+            deviceMap.put(device.getMac(), device);
+        }
+        return deviceMap;
+    }
+
+    private Map<String, UniFiClient> getClients() throws UniFiException {
+        Map<String, UniFiClient> clientMap = new HashMap<>();
+        Collection<UniFiSite> sites = sitesCache.values();
+        for (UniFiSite site : sites) {
+            Map<String, UniFiClient> siteClientMap = getClients(site);
+            clientMap.putAll(siteClientMap);
+        }
+        return clientMap;
+    }
+
+    private Map<String, UniFiClient> getClients(UniFiSite site) throws UniFiException {
+        Map<String, UniFiClient> clientMap = new HashMap<>();
+        UniFiClient[] clients = controller.getClients(site);
+        logger.debug("Found {} UniFi Client(s):", clients.length);
+        for (UniFiClient client : clients) {
+            client.setDevice(devicesCache.get(client.getDeviceMac()));
+            logger.debug("  {}", client);
+            cachePut(clientMap, client);
+        }
+        return clientMap;
+    }
+
+    private Map<String, UniFiClient> getInsights() throws UniFiException {
+        Map<String, UniFiClient> insightsMap = new HashMap<>();
+        Collection<UniFiSite> sites = sitesCache.values();
+        for (UniFiSite site : sites) {
+            Map<String, UniFiClient> siteInsightsMap = getInsights(site);
+            insightsMap.putAll(siteInsightsMap);
+        }
+        return insightsMap;
+    }
+
+    private Map<String, UniFiClient> getInsights(UniFiSite site) throws UniFiException {
+        Map<String, UniFiClient> insightsMap = new HashMap<>();
+        UniFiClient[] clients = controller.getInsights(site);
+        logger.debug("Found {} UniFi Insights(s):", clients.length);
+        for (UniFiClient client : clients) {
+            logger.debug("  {}", client);
```

Comment:
These debug messages with only a space and the argument can be hard to understand if the value is unexpected. Maybe put in some context?

Reply:
I'm not sure what else to do here. I'm simply logging a list of all clients returned by the controller. The reason for the two spaces is simply to indent the output to make it a little easier to read in the console.

How would you suggest I change this?

Reply:
Maybe you could move the output to the logger above, something like „log.debug(„Found {} UniFi Insight(s): {}“, clients.length, clients);“

I think what @Hilbrand means is that there is no guarantee that there are no log lines from other modules in between and it might be hard to see that these lines belong together.

Reply:
I understand now... thanks for the clarification.

I merged the log statements and added a lazy formatter to the arrays as to not do the string concatenation unless the debug log level was active.